src/tests/error_throwing_tests.rs

```diff
@@ -0,0 +1,23 @@
+use crate::utils::{js_typed_error, ErrorType};
+
+#[napi]
+/// Test function that throws specified error with a custom message.
+pub fn throw_test_error(error_type: String, custom_message: Option<String>) -> napi::Result<()> {
```

Comment:
Those values are passed through napi so they cannot be of ``&str`` or generic type.

Reply:
1. Hmm, but IIRC we can accept `&T` in Rust functions called from JS through napi. At least I saw `Vec<&T>`, so why `&T` would not work?
2. If `&str` is problematic, then maybe `&String`? Just to avoid allocations.

Reply:
1. I don't remember any place in code where we accept generic types (or vector of generic types). From my testing any attempt to add generic to napi exposed functions results in CE

Reply:
> 1. I don't remember any place in code where we accept generic types (or vector of generic types). From my testing any attempt to add generic to napi exposed functions results in CE

That's correct, what I meant by `T` was some specific type `T`, not a generic parameter. Here, I'd just like to pass `&str` or `&String`.

Reply:
Passing a reference instead of a value would mean we have a reference to relevant string in JS and there is not enough guarantees of its lifetime. From what i read there is also an issue that Rust and JS have different ways of encoding strings (UTF-8 in Rust and UTF-16 in JS).

Reply:
So it means this just doesn’t compile?

Reply:
It throws a compilation error because ``&String`` and ``&str`` do not implement FromNapiRef trait.

Reply:
We could do ``Vec<&T>`` in other places because that specific ``T`` implemented this trait.

Reply:
> Passing a reference instead of a value would mean we have a reference to relevant string in JS and there is not enough guarantees of its lifetime.

I don't think that lifetime is an issue here.

> From what I read there is also an issue that Rust and JS have different ways of encoding strings (UTF-8 in Rust and UTF-16 in JS).

This explains everything... Gosh, UTF-16 is so useless... (but still Java, JS, and Windows keep using it for some reason...)